text/0000-const-generics.md

```diff
@@ -0,0 +1,287 @@
+- Feature Name: const_generics
+- Start Date: 2017-05-01
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow types to be generic over constant values; among other things this will
+allow users to write impls which are abstract over all array types.
+
+# Motivation
+[motivation]: #motivation
+
+Rust currently has one type which is parametric over constants: the built-in
+array type `[T; LEN]`. However, because const generics are not a first class
+feature, users cannot define their own types which are generic over constant
+values, and cannot implement traits for all arrays.
+
+As a result of this limitation, the standard library only contains trait
+implementations for arrays up to a length of 32; as a result, arrays are often
```

Comment:
Note: not just the standard library itself has this limitation, but also libraries like [serde have it](https://docs.serde.rs/serde/trait.Deserialize.html).